odoo/models.py

```diff
@@ -2308,7 +2308,7 @@ def _read_group_raw(self, domain, fields, groupby, offset=0, limit=None, orderby
             assert gb in self._fields, "Unknown field %r in 'groupby'" % gb
             assert self._fields[gb].base_field.groupable, (
                 f"Field {gb} of model {self._name} is not a stored field, only stored fields "
-                "(regular or many2many) are valid for the 'groupby' parameter."
+                "(regular, related or many2many) are valid for the 'groupby' parameter."
```

Comment:
:point_up: 

My previous variant was as following:

            # for non-stored related field we do a quick check here and deep check in _inherits_join_calc
           assert (gb_field.store or gb_field.related and gb_field.related_field.store) and gb_field.column_type, "Fields in 'groupby' must be regular database-persisted fields, or function fields with store=True, or related field without one2many element in the chain"